odxtools/identdesc.py

```diff
@@ -0,0 +1,54 @@
+# SPDX-License-Identifier: MIT
+from dataclasses import dataclass
+from typing import Any
+from xml.etree import ElementTree
+
+from .exceptions import odxrequire
+from .odxdoccontext import OdxDocContext
+from .odxlink import OdxLinkDatabase, OdxLinkId
+from .snrefcontext import SnRefContext
+
+
+@dataclass(kw_only=True)
+class IdentDesc:
+    diag_comm_snref: str
+    ident_if_snref: str
+
+    # exactly one of the two attributes below must be defined
+    out_param_if_snref: str | None = None
+    out_param_if_snpathref: str | None = None
+
+    @staticmethod
+    def from_et(et_element: ElementTree.Element, context: OdxDocContext) -> "IdentDesc":
+        diag_comm_snref = odxrequire(
+            odxrequire(et_element.find("DIAG-COMM-SNREF")).attrib.get("SHORT-NAME"))
+        ident_if_snref = odxrequire(
+            odxrequire(et_element.find("IDENT-IF-SNREF")).attrib.get("SHORT-NAME"))
+        out_param_if_snref = None
+        if (out_param_if_snref_elem := et_element.find("OUT-PARAM-IF-SNREF")) is not None:
+            out_param_if_snref = odxrequire(out_param_if_snref_elem.attrib.get("SHORT-NAME"))
+
+        out_param_if_snpathref = None
+        if (out_param_if_snpathref_elem := et_element.find("OUT-PARAM-IF-SNPATHREF")) is not None:
+            out_param_if_snpathref = odxrequire(
+                out_param_if_snpathref_elem.attrib.get("SHORT-NAME-PATH"))
+
+        return IdentDesc(
+            diag_comm_snref=diag_comm_snref,
+            ident_if_snref=ident_if_snref,
+            out_param_if_snref=out_param_if_snref,
+            out_param_if_snpathref=out_param_if_snpathref,
+        )
+
+    def _build_odxlinks(self) -> dict[OdxLinkId, Any]:
+        odxlinks: dict[OdxLinkId, Any] = {}
+
+        return odxlinks
```

Comment:
Objects that don't participate in the link resolution, should not have such methods, they waste CPU and RAM without benefit.

Reply:
some objects which do not have an ODX ID still "own" objects which do. I'd like to always have the full set of the link resolution functions or none at all... (the performance penalty of calling empty functions is negligible IMO.)

Reply:
(in this case the SNREFs might need to be resolved in the future...)

Reply:
given enough objects, even calling an empty function have a cost

Reply:
yes, but compared to the cost of parsing and internalizing the XML, it is pretty negligible. I don't think that having to wait a second more for a dataset that takes two minutes to load makes a big difference in practice...